src/Symfony/Component/Config/FileLocator.php

```diff
@@ -42,11 +42,15 @@ public function locate($name, $currentPath = null, $first = true)
         }
 
         if ($this->isAbsolutePath($name)) {
-            if (!file_exists($name)) {
+            if (array() === $files = glob($name, defined('GLOB_BRACE') ? GLOB_BRACE : 0)) {
                 throw new FileLocatorFileNotFoundException(sprintf('The file "%s" does not exist.', $name));
             }
 
-            return $name;
+            if ($first) {
```

Comment:
Actually I think we should keep the original behavior as it was (just invert the `file_exists` check), then add the glob as an extra step.
Otherwise if you do something like `/etc/myapp/*.yml`, you would expect an array, so it might be weird to get a string if the glob only matches a single file